File: readability/readability.c
#include <cs50.h>
#include <stdio.h>
#include <string.h>
#include <ctype.h>
#include <math.h>


float count_letters(string text);
float count_words(string text);
float count_sentences(string text);
int grade(float words, float letters, float sentences);


int main(void)
{
    string text = get_string("Your text: \n");
    int g = grade(count_words(text), count_letters(text), count_sentences(text));
    // Checking if the value is greater then 16
    if (g >= 16)
    {
        printf("Grade 16+\n");
    }
    // Checking if value is lower then 1
    else if (g < 1)
    {
        printf("Before Grade 1\n");
    }
    else
    {
        printf("Grade %i\n", g);
    }
}


float count_letters(string text)
{
    // Counting alphabetical letters in text
    float letters = 0;
    for (int i = 0; i < strlen(text); i++)
    {
        // Checking if value is alphabetical
        if (isalpha(text[i]))
        {
            letters++;
        }
    }
    // Returns the amount of letters in text
    return letters;
}

float count_words(string text)
{
    // Counting words by finding space in text
    float words = 1;
    for (int i = 0; i < strlen(text); i++)
    {
        if (isspace(text[i]))
        {
            words++;
        }
    }
    // Returns the amount of words in text
    return words;
}

float count_sentences(string text)
{
    // Counting sentences by finding . ! ? in text
    float sentences = 0;
    for (int i = 0; i < strlen(text); i++)
    {
        // Looking for . ? !
        if ((text[i] == '.') || (text[i] == '!') || (text[i] == '?'))
        {
            sentences++;
        }
    }
    // Returning the float value
    return sentences;
}

// Function that is taking all floats and doing math
int grade(float words, float letters, float sentences)
{
    // Doing math to round the grade
    float l = (letters / words * 100);
    float s = (sentences / words * 100);
    // Returning the rounded int value
    return round(0.0588 * l - 0.296 * s - 15.8);
}

File: sentimental-readability/readability.py
# Main function
def main():
    # Get text
    text = input("Text: ")
    # Declare variables
    letters = count_letters(text)
    words = count_words(text)
    sentences = count_sentences(text)
    L = 100 * (letters / words)
    S = 100 * (sentences / words)
    grade = round(0.0588 * L - 0.296 * S - 15.8)
    # Print grade
    if grade < 1:
        print("Before Grade 1")
    elif grade > 16:
        print("Grade 16+")
    else:
        print(f"Grade {grade}")


# Count number of letters in text
def count_letters(text):
    # Set counter
    countletter = 0
    for c in text:
        # checking if character is alphabetical
        if (c.isalpha()) == True:
            # Add 1 to counter
            countletter += 1
    return countletter


# Count number of words in text
def count_words(text):
    # Set counter
    countword = 1
    for c in text:
        space = ' '
        # counting words by counting spaces
        if c == space:
            # Add 1 to counter
            countword += 1
    return countword


# Count number of sentences in text
def count_sentences(text):
    # Set counter
    countsen = 0
    for c in text:
        exclamation = '!'
        period = '.'
        questionmark = '?'
        # If character in text is an exclamation mark, question mark or period
        if c == exclamation or c == period or c == questionmark:
            # Add 1 to counter
            countsen += 1
    return countsen


main()